src/plugins/visualizations/server/migrations/visualization_saved_object_migrations.ts

```diff
@@ -1182,3 +1192,55 @@ export const visualizationSavedObjectTypeMigrations = {
   '7.17.0': flow(addDropLastBucketIntoTSVBModel714Above),
   '8.0.0': flow(removeMarkdownLessFromTSVB),
 };
+
+const mergeSavedObjectMigrationMaps = (
+  obj1: SavedObjectMigrationMap,
+  obj2: SavedObjectMigrationMap
+): SavedObjectMigrationMap => {
+  const customizer = (objValue: SavedObjectMigrationFn, srcValue: SavedObjectMigrationFn) => {
+    if (!srcValue || !objValue) {
+      return srcValue || objValue;
+    }
+    return (state: SavedObjectUnsanitizedDoc, context: SavedObjectMigrationContext) =>
+      objValue(srcValue(state, context), context);
+  };
+
+  return mergeWith({ ...obj1 }, obj2, customizer);
+};
```

Comment:
I'd like to relocate this function so that other plugins have access. It feels "core"-ish to me since that's where the other saved object stuff is defined. Any thoughts on where this might go?

cc: @elastic/kibana-core

Reply:
Are we already aware of other plugins that would benefit from using this utility function? If not, I'd would rather keep it in `visualizations` for now. Else it can be added as a `utils.ts` file in `src/core/server/saved_objects/migrations` and exported from the entrypoint.

Reply:
we expose a similar utility here: https://github.com/elastic/kibana/blob/main/src/plugins/kibana_utils/common/persistable_state/merge_migration_function_map.ts

would it make sense to rather update that one to serve this usecase as well ?

Reply:
> would it make sense to rather update that one to serve this usecase as well ?

Would probably, however this function is in a `common` module, so you wouldn't be allowed to import the `SavedObjectMigrationMap` type which is in a `server` module I believe?

Reply:
@pgayvallet 
> Are we already aware of other plugins that would benefit from using this utility function?

Good question—yes, Lens needs it.

@ppisljar 
> would it make sense to rather update that one to serve this usecase as well?

This is a good idea. In fact, it's the strategy I originally pursued, but I ran into trouble since the `MigrateFunction` type only accepts a single state parameter, but the `SavedObjectMigrationFn` accepts a context object as a second parameter that has to be passed along during the [merge operation](https://github.com/elastic/kibana/pull/123005/files#diff-1a8bd3dafa53191c87e0c6fe9ca79527dd30e21bb9b4acfe4b4551ef794d341eR1205). I did not know about [this restriction](https://github.com/elastic/kibana/pull/123005#discussion_r785733066) either.

Given the obstacles, I suggest that I keep them separate for this PR, but that I open an issue to revisit.

Reply:
@pgayvallet I followed your suggestion and added a `utils.ts` file at that path. However, I am seeing an error "Cannot find module 'kibana/server'" when I run Kibana server. It originates from [this import statement](https://github.com/elastic/kibana/pull/123005/files#diff-1a8bd3dafa53191c87e0c6fe9ca79527dd30e21bb9b4acfe4b4551ef794d341eR11) but I cannot see anything the matter with what I'm doing. Any idea why this might be happening?

Reply:
Concrete imports needs to be imported via fullpaths. Only virtuals (types) imports can be resolved via `kibana/server`.

You'll need to import from `.........../src/core/server`

Reply:
@pgayvallet thank you, that worked! However I've run into another issue building ts refs: https://github.com/elastic/kibana/issues/123430. I thought it seemed like an issue for the core team, but let me know if I should talk to someone else.